### ISSUE TITLE
add projects section and create a very pretty project card

### ISSUE THREAD

**Issue by Majestic9169** [Majestic9169/metakgp-website#3]
I really like the cards they have [here](https://hackclub.com/)

![image](https://github.com/user-attachments/assets/3e44fe8d-591f-4e33-84f8-d9fbaf61fda8)

something unique like this for the most prominent projects would be awesome

**Comment by ShrushtiDesai** [comment]
Hi @Majestic9169,
I’m interested in working on this issue.

**Comment by Majestic9169** [comment]
All yours, looking forward to a good design!

**Comment by ShrushtiDesai** [comment]
Hi @Majestic9169,
I have a few questions and suggestions regarding the implementation:
- I’m referring to this [**page**](https://metakgp.github.io/) for the list of projects that need to be displayed on the Projects page.
- Regarding the **HeaderSection**, I think it will be present on every page. So If it’s common across all pages, we could include it in the **App.tsx** file so it won't be required to import it in every page. 
- We can use **NavLink** instead of **Route** for navigation since it provides the **isActive** state, making it easier to highlight the active tab.

For the project cards design, based on the reference [link](https://hackclub.com/) you provided:
- The **stars** and **GitHub link** could be placed in the **top-right corner**.
- The **project name** could be on the **left side**, with a **short description** below it.
- Regarding the **background image** for the cards, the reference link uses an image related to the project. Do you have these images available, or should I leave them blank or create placeholders for now?

Let me know your thoughts and suggestions!

**Comment by ShrushtiDesai** [comment]
Hi @Majestic9169,
I have a few questions and suggestions regarding the implementation:
- I’m referring to this [page](https://metakgp.github.io/) for the list of projects that need to be displayed on the Projects page.
- Regarding the **HeaderSection**, I think it will be present on every page. If it’s common across all pages, then instead of importing the header in each individual page, we can create a **Layout** component that will contain the **HeaderSection** as well as a **children prop** where the other pages will be rendered. Then, in **index.tsx**, we can wrap the routes inside this **Layout** component. This will ensure that the **header** appears on every page.
- We can use **NavLink** instead of **a** tags for navigation. As it provides the **isActive** state, which makes it easier to apply active styles and highlight the current tab.
- we can make the **logo** in the **HeaderSection** to navigate to the home (/) page.

For the project cards design, based on the reference [link](https://hackclub.com/) you provided:
- The **stars** and **GitHub link** could be placed in the **top-right corner**.
- The **project name** could be on the **left side**, with a **short description** below it.
- The **background image** for the cards, the reference link uses an image related to the project. Do you have these images available, or should I leave them blank or create placeholders for now?

Let me know your thoughts and suggestions!

**Comment by Majestic9169** [comment]
Please look at the more active projects on metakgp's github instead of following that link

As of now there are no specific background images, feel free to use some freely available backgrounds

Your other points are valid, feel free to create a pr

**Comment by ShrushtiDesai** [comment]
Hi @Majestic9169,
I’ve designed the project cards, and they currently look like this:
![gitss](https://github.com/user-attachments/assets/637ac941-7708-4054-b484-6c904e4f5a6a)
To reduce repetitive code, I’ve stored all the project information (like background image, title image, stars, GitHub link, and website link) in an array of objects. However, there’s an issue: not all projects have a website link since some of them aren’t deployed. This creates inconsistencies. I’m considering switching to separate and repeated code for each card instead. What do you think about this approach?

Another question: how can I dynamically fetch the GitHub stars for each repository?

I’ve been following the MetaKGP GitHub repositories for projects [link](https://github.com/metakgp), but there are over 50 repositories in total. Which ones should I prioritize for the projects page?

Let me know your thoughts and suggestions!

**Comment by Waqibsk** [comment]
Will all the project cards have the same design? @ShrushtiDesai

**Comment by ShrushtiDesai** [comment]
Yes @Waqibsk, 
As i was rendering the data from the array of objects using loop in html. The background-image, title, button links, stars, and Github link will be different. 
Overall the position of the items remains the same.
But since not all projects have been deployed so it would lead to inconsistencies in the button link for "Go to Website".
So i was considering of writing seperate code for each Project card if there is no solution.

**Comment by Waqibsk** [comment]
Ok then ,i think we can use the github api mentioned in PR :#9  ...after a few changes like ...

1.Sort the repos by their stars or recent activity and get the top 5 or 6 repos
2.Then we can use your project card template and render all the projects 


This way, we can retrieve all the required data from GitHub directly without manually entering it. For the "Go to website" button, we can check whether a hosted page exists or not. 
However, if we follow this approach,

1. We won't be able to display any project-specific background images.
2. As the description will be taken from GitHub, it may lead to inconsistencies.

**Comment by ShrushtiDesai** [comment]
@Waqibsk, that a good idea!
but we require the background images and description if we have to follow the design suggested for this page by @Majestic9169 in the first comment.
Maybe we can combine data from Github and manually add data which is not obtained from Github like the Background image and project description?
i am not well versed in react and typescript so not sure if its possible.
let me know your thoughts on it.

**Comment by Waqibsk** [comment]
I would like to hear @Majestic9169's thoughts on this:

1.Is it appropriate to use random images for backgrounds, as shown below..?


<div>
  <img src="https://img.freepik.com/free-vector/copy-space-dark-background-assortment-geometrical-shapes_23-2148383737.jpg?t=st=1735993392~exp=1735996992~hmac=bad2147be3d9b3bff88dab07efbd064a95cb4678eeaa8774fd11041a4d481860&w=996" width="300">

  <img src="https://img.freepik.com/free-vector/abstract-colorful-background_23-2148929192.jpg?t=st=1735993973~exp=1735997573~hmac=c242fba358ae2b2b56c4eb2ad16b778d6e8b3614da9ad0e36dc33f110406c7b0&w=996" width="300">

  <img src="https://img.freepik.com/premium-vector/wallpaper-background-design-vector-illustration_1021543-757.jpg?w=996" width="300">
 <img src="https://img.freepik.com/free-vector/hand-drawn-flat-abstract-shapes-background_23-2149083556.jpg?t=st=1735994829~exp=1735998429~hmac=03b48cd4e1136795cde274d63dd6d54c2bb13955a1dbb7802b6f46c8970baf9d&w=900" width="300">
</div>


2 .Is it ok to use the project description directly from metakgp GitHub?"

**Comment by Ananya-m0140** [comment]
@ShrushtiDesai @Waqibsk 
Regarding the above comments,
- We would like to have different background images for the different  projects. 
- Instead of some random images, put project specific backgrounds (any attractive looking project related image will do, must not be too informal).
- You can directly use the project description from metakgp Github .
- As of which repos to choose, choose the ones with most stars and the most active ones. 15 to 20 of them are enough.

**Comment by Majestic9169** [comment]
btw i like the backgrounds @Waqibsk shared as long as they match the vibe of the project

**Comment by Waqibsk** [comment]
Maybe we can merge the project card template provided by @ShrushtiDesai and work on the styling and data fetching for the cards in a separate issue. This way, we will have a base template ready to test things out

**Comment by Majestic9169** [comment]
i agree, @ShrushtiDesai could you make a draft pr with your progress?

**Comment by ShrushtiDesai** [comment]
@Majestic9169, @Waqibsk
I'll create the PR with my progress.
